Source/missiles.cpp

```diff
@@ -3619,20 +3615,16 @@ void MI_Apoca(Missile &missile)
 				continue;
 			if (gbIsHellfire && !LineClearMissile(missile.position.tile, { k, j }))
 				continue;
+
+			int id = missile._misource;
 			AddMissile({ k, j }, { k, j }, Players[id]._pdir, MIS_BOOM, TARGET_MONSTERS, id, missile._midam, 0);
-			exit = true;
-		}
-		if (!exit) {
-			missile.var4 = missile.var6;
+			missile.var2 = j - 1;
+			missile.var4 = k;

```

Comment:
```suggestion
			missile.var2 = j;
			missile.var4 = k + 1;
```
With this changes you should have the old behavior (see other comment).